### PR TITLE
[NO-TICKET] - Fix download pdf attachments

### PR DESCRIPTION
**Pull request by bamaro-dev**

### CODE REVIEW

File: Billing.API/Services/SapApi/SapApiService.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                                    Password = serviceSetting.Password,
+                                    UserName = serviceSetting.UserName
+                                }),
+                            Encoding.UTF8,
+                            "application/json"),
+                        Method = HttpMethod.Post
+                    });
+                    sapResponse.EnsureSuccessStatusCode();
+
+                    var sessionTimeout = JObject.Parse(await sapResponse.Content.ReadAsStringAsync());
```
Comment (nreal) [inline Billing.API/Services/SapApi/SapApiService.cs:68]: I think that this variable is not used never

File: Billing.API/Services/SapApi/SapApiService.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                                {
+                                    CompanyDB = serviceSetting.CompanyDB,
+                                    Password = serviceSetting.Password,
+                                    UserName = serviceSetting.UserName
+                                }),
+                            Encoding.UTF8,
+                            "application/json"),
+                        Method = HttpMethod.Post
+                    });
+                    sapResponse.EnsureSuccessStatusCode();
```
Comment (nreal) [inline Billing.API/Services/SapApi/SapApiService.cs:66]: I think that this line should be remove

**Comment by nreal** [comment]
👍

File: Billing.API/Services/SapApi/SapApiService.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+            var cookies = await StartSession(serviceSetting);
+            message.Headers.Add("Cookie", cookies.B1Session);
+            message.Headers.Add("Cookie", cookies.RouteId);
+
+            var client = _httpClientFactory.CreateClient();
+            return await client.SendAsync(message);
+        }
+
+        public async Task<SapLoginCookies> StartSession(SapServiceConfig serviceSetting)
```
Comment (nreal) [inline Billing.API/Services/SapApi/SapApiService.cs:45]: I think that this method should be **private**

File: Billing.API/Services/SapApi/SapApiService.cs (hunk 1 above)
Comment (bamaro-dev) [reply Billing.API/Services/SapApi/SapApiService.cs:68]: Actually, it is used to set `SessionEndAt` propierty of Sap cookies.

File: Billing.API/Services/SapApi/SapApiService.cs (hunk 2 above)
Comment (bamaro-dev) [reply Billing.API/Services/SapApi/SapApiService.cs:66]: Maybe you are right, but the problem is that if we remove this line, and the Login doesnt response, a few lines later when it try to use sapResponse.Headers, it is going to crash.....Am I right?

File: Billing.API/Services/SapApi/SapApiService.cs (hunk 3 above)
Comment (bamaro-dev) [reply Billing.API/Services/SapApi/SapApiService.cs:45]: You're right.

**Pull request merged**